packages/docusaurus/lib/client/prefetch.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const support = function(feature) {
+  if (typeof document === `undefined`) {
+    return false;
+  }
+  const fakeLink = document.createElement(`link`);
+  try {
+    if (fakeLink.relList && typeof fakeLink.relList.supports === `function`) {
+      return fakeLink.relList.supports(feature);
+    }
+  } catch (err) {
+    return false;
+  }
+  return false;
+};
+
+const linkPrefetchStrategy = function(url) {
+  return new Promise((resolve, reject) => {
+    if (typeof document === `undefined`) {
+      reject();
+      return;
+    }
+
+    const link = document.createElement(`link`);
+    link.setAttribute(`rel`, `prefetch`);
+    link.setAttribute(`href`, url);
+
+    link.onload = resolve;
+    link.onerror = reject;
+
+    const parentElement =
+      document.getElementsByTagName(`head`)[0] ||
+      document.getElementsByName(`script`)[0].parentNode;
+    parentElement.appendChild(link);
+  });
+};
+
+const xhrPrefetchStrategy = function(url) {
+  return new Promise((resolve, reject) => {
+    const req = new XMLHttpRequest();
```

Comment:
Use `fetch` instead

Reply:
Actually this code is copied from Gatsby https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/prefetch.js

I think it's battle tested 😆